storops/unity/resource/user_quota.py

```diff
@@ -0,0 +1,128 @@
+# coding=utf-8
+# Copyright (c) 2015 EMC Corporation.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from __future__ import unicode_literals
+
+import logging
+
+import storops
+from storops.unity.resource import UnityResource, UnityResourceList
+from storops.unity.client import UnityClient
+
+__author__ = 'Rajendra Indukuri'
+
+log = logging.getLogger(__name__)
+
+
+class UnityUserQuota(UnityResource):
+    """
+    Support CRUD operations for user_quota
+
+    Operations supported:
+    create: Create user_quota on a file_system/tree_quota
+    modify: Modify user_quota on a file_system/tree_quota
+    """
+    @classmethod
+    def create(cls, cli, file_system_id=None, tree_quota_id=None,
+               hard_limit=None, soft_limit=None, uid=None,
+               unix_name=None, win_name=None):
+        """
+        Creates user_quota on the speficied filesystem or tree quota
+        :param file_system_id: This is an optional parameter which if provided
+            will create user_quota on the file_system specified. file_system_id
+            and tree_quota_id cannot be given together
+        :param tree_quota_id: This is an optional parameter which if provided
+            will create user_quota on the tree_quota specified. file_system_id
+            and tree_quota_id cannot be given together
+        :param hard_limit: sets hard_limit  on the file_system for the user
+            specified
+        :param soft_limit: sets soft_limit  on the file_system for the user
+            specified
+        :param uid: This is an optional parameter to specify user. Either of
+            uid/unix_name/win_name should be provided
+        :param unix_name: This is an optional parameter to specify user. Either
+            of uid/unix_name/win_name should be provided
+        :param win_name:This is an optional parameter to specify user. Either
+            of uid/unix_name/win_name should be provided
+        :return: created user quota.
+        """
+        fs_clz = storops.unity.resource.filesystem.UnityFileSystem
+        file_system = fs_clz.get(cli, file_system_id).verify()
+        user_quota_param = cls.prepare_user_quota_create_parameters(
+            file_system, tree_quota_id, hard_limit, soft_limit, uid,
```

Comment:
This is the definition of prepare_user_quota_create_parameters:
```
    def prepare_user_quota_create_parameters(
                  file_system, tree_quota, hard_limit,
                  soft_limit, uid, unix_name, win_name):
```
Will the tree_quota parameter accept an object or an ID string? It should be an treeQuota object, right? So the treeQuota should be passed to the invocation of cls.prepare_user_quota_create_parameters, right?

Reply:
@yong-huang Ya true.. here it will be a tree_quota object.. this is just a place holder and this is implemented as part of my next PR which is WIP... for more details, pls check the below branch if you want to take a look:
https://github.com/rajendraindukuri/storops/tree/support-tree-quota-management

Reply:
It is a little confused if the prepare_user_quota_create_parameters accept tree_quota object, but tree_quota_id is passed to.

My concern is the request body should be built like:
```
  "treeQuota": {
    "id": "tree_quota_id"
  },
```
But for now, the request body is:
```
  "treeQuota": "tree_quota_id"
```

Did you test this parameter? Seems it is not covered by UT, now the tested request is:
```
    {
      "url": "/api/types/userQuota/instances?compact=True",
      "body": {
        "filesystem": {
          "id": "fs_2"
        },
        "softLimit": 3221225472,
        "hardLimit": 9663676416,
        "uid": 3
      },
      "response": "create_user_quota_1_success.json"
    },
```

Reply:
Hi @yong-huang , Whatever you mentioned perfectly right. As I said, tree_quota is just  a placeholder here... since we are not supporting tree_quota as part of this PR, cannot validate it. So just put it as a placeholder and yes there is no UT for this tree_quota in this PR as nothing related to this is implemented

Covering this as part of my next PR :
![image](https://user-images.githubusercontent.com/82365588/115507102-4f5b1f80-a299-11eb-9e02-09f0dc644c0f.png)
UT s also will be covered for the same in my next PR